packages/filler/src/services/CacheService.ts

```diff
@@ -0,0 +1,118 @@
+interface GasEstimateCache {
+	fillGas: string
+	postGas: string
+	timestamp: number
+}
+
+interface SwapCall {
+	to: string
+	data: string
+	value: string
+}
+
+interface SwapOperationsCache {
+	calls: SwapCall[]
+	totalGasEstimate: string
+	timestamp: number
+}
+
+interface CacheData {
+	gasEstimates: Record<string, GasEstimateCache>
+	swapOperations: Record<string, SwapOperationsCache>
+}
+
+export class CacheService {
```

Comment:
I don't think this is neccessary

Reply:
This has improved execution speed. I added it so that `executeOrder` could leverage from all the contract reads, simulations already done by `calculateProfitability` section.

Reply:
I can see why it can be helpful. We do a lot of eth_calls. Since these are network calls, they increase the chances of failure, which could cause orders not to be filled. Also, these estimates won't change drastically(they are not dependent on gas price), so it might be beneficial to do them once.